app/experimenter/legacy-ui/rapid/components/experiments/ResultsTable.tsx

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+
+import ResultsRow from "experimenter-rapid/components/experiments/ResultsRow";
+import { RESULT_COLUMN_TIPS } from "experimenter-rapid/components/experiments/tooltips";
+import { featureOptions } from "experimenter-rapid/components/forms/ExperimentFormOptions";
+import { AnalysisPoint, ExperimentData } from "experimenter-types/experiment";
+
+enum Significance {
+  POSITIVE,
+  NEGATIVE,
+  NEUTRAL,
+}
+
+const Metric = {
+  RETENTION: "retained",
+  SEARCH: "search_count",
+  USER_COUNT: "identity",
+};
+
+const getResultMetrics = (
+  featureData: Array<string>,
+): Array<{ value: string; name: string; tooltip: string }> => {
+  // This is used in conjunction with resultsData as an ordered
+  // list of metrics to display in the table from left to right
+  const resultsMetricsList = [
+    {
+      value: Metric.RETENTION,
+      name: "2-Week Browser Retention",
+      tooltip: RESULT_COLUMN_TIPS.RETENTION,
+    },
+    {
+      value: Metric.SEARCH,
+      name: "Daily Mean Searches Per User",
+      tooltip: RESULT_COLUMN_TIPS.SEARCH,
+    },
+    {
+      value: Metric.USER_COUNT,
+      name: "Total Users",
+      tooltip: RESULT_COLUMN_TIPS.USER_COUNT,
+    },
+  ];
+
+  const featureNameMappings = featureOptions.reduce((acc, featureRow) => {
+    acc[featureRow.value] = featureRow["name"];
+    return acc;
+  }, {});
+  featureData.forEach((feature) => {
+    const featureMetricID = `${feature}_ever_used`;
+    resultsMetricsList.unshift({
+      value: featureMetricID,
+      name: `${featureNameMappings[feature]} Conversion`,
+      tooltip: RESULT_COLUMN_TIPS.CONVERSION,
+    });
+  });
+
+  return resultsMetricsList;
+};
+
+const computeSignificance = (lower: number, upper: number): Significance => {
+  if (lower < 0 && upper < 0) {
+    return Significance.NEGATIVE;
+  } else if (lower > 0 && upper > 0) {
+    return Significance.POSITIVE;
+  } else {
+    return Significance.NEUTRAL;
+  }
+};
+
+const getResultsData = (data: ExperimentData) => {
```

Comment:
Could this processing be done on the backend?

Reply:
Technically, yes. However, I would prefer postponing moving too much to the backend just yet as there will be more data processing done for the other pieces of the dashboard and I don't yet know what that will all look like. I think once it's all in there, it will be easier to polish which data is processed where.

I would prefer to do this iteratively as more parts of the dashboard are implemented.